lib/AST/Module.cpp

```diff
@@ -2246,6 +2246,48 @@ SPIGroupsRequest::evaluate(Evaluator &evaluator, const Decl *decl) const {
   return ctx.AllocateCopy(spiGroups.getArrayRef());
 }
 
+LibraryLevel ModuleDecl::getLibraryLevel() const {
+  return evaluateOrDefault(getASTContext().evaluator,
+                           ModuleLibraryLevelRequest{this},
+                           LibraryLevel::Other);
+}
+
+LibraryLevel
+ModuleLibraryLevelRequest::evaluate(Evaluator &evaluator,
+                                    const ModuleDecl *module) const {
+  auto &ctx = module->getASTContext();
+
+  /// Is \p path from System/Library/PrivateFrameworks/?
+  auto fromPrivateFrameworks = [&](StringRef path) -> bool {
+    auto sep = llvm::sys::path::get_separator();
+    auto privateFrameworksPath = ctx.SearchPathOpts.SDKPath +
+      sep + "System" + sep + "Library" + sep + "PrivateFrameworks" + sep;
```

Comment:
Nit: start with `llvm::Twine(ctx.SearchPathOpts.SDKPath)` here.

Reply:
While applying this I came across `llvm::sys::path::append` which simplifies greatly this piece of code.